test/GetIntrinsic.js

```diff
@@ -34,6 +34,36 @@ test('throws', function (t) {
 		'"just a dot" intrinsic throws a syntax error'
 	);
 
+	t['throws'](
+		function () { GetIntrinsic('%String'); },
+		SyntaxError,
+		'Leading % without trailing % throws a syntax error'
+	);
+
+	t['throws'](
+		function () { GetIntrinsic('String%'); },
+		SyntaxError,
+		'Trailing % without leading % throws a syntax error'
+	);
+
+	t['throws'](
+		function () { GetIntrinsic('String["prototype"]'); },
+		SyntaxError,
+		'Dynamic property access is disallowed for intrinsics (double quote)'
+	);
+
+	t['throws'](
+		function () { GetIntrinsic("String['prototype']"); },
+		SyntaxError,
+		'Dynamic property access is disallowed for intrinsics (single quote)'
+	);
```

Comment:
Was this actually allowed previously, with strings?

Dynamic access is *required* for symbol lookups, so if anything's disallowed inside square brackets it should only be string properties.

Reply:
This didn’t support symbol properties.

If you did: `GetIntrinsic('Array.prototype[Symbol.iterator]')`, you’d get a non‑functional result (`undefined`) anyway.

Reply:
if it was never possible to use brackets before, then i'm not sure why we'd need to add an exception since nobody could be relying on it working.

Reply:
If I remove this check, then it would consider `GetIntrinsic('Array.prototype[push]')` as equal to `GetIntrinsic('Array.prototype.push')`.

Reply:
I definitely want to fix the bug that allows `'Array.prototype[push]'`.